### PR TITLE
feat: add minifile

### PR DESCRIPTION
**Pull request by sandyzhou**

### CODE REVIEW

File: storage/minifile/mimifile.go [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+
+	mf.lock.RLock()
+	defer mf.lock.RUnlock()
+
+	mf.namedLock.Lock(key)
+	defer mf.namedLock.UnLock(key, false)
+
+	name := filepath.Join(mf.path, key)
+
+	if err := ioutil.WriteFile(name, value, 0644); err != nil {
```
Comment (Lizen0512) [inline storage/minifile/mimifile.go:65]: please add a computed checksum to ensure the consistency between write and read of files

File: storage/minifile/mimifile.go [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+	val, err := ioutil.ReadFile(name)
+	if err != nil && isNoFileError(err) {
+		_ = mf.namedLock.RUnLock(key, true)
+		return nil, nil
+	}
+
+	_ = mf.namedLock.RUnLock(key, false)
+
+	crc := make([]byte, 4)
+	binary.LittleEndian.PutUint32(crc, util.NewCRC(val[:len(val)-4]).Value())
```
Comment (Lizen0512) [inline storage/minifile/mimifile.go:120]: Please check that the length of val is greater than 4

File: storage/minifile/mimifile.go [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+	"sync/atomic"
+
+	"github.com/prometheus/tsdb/fileutil"
+	"github.com/syndtr/goleveldb/leveldb/util"
+)
+
+type MiniFile struct {
+	path         string
+	instanceLock fileutil.Releaser // File-system lock to prevent double opens
+	namedLock    *NamedLock
```
Comment (Lizen0512) [inline storage/minifile/mimifile.go]: It seems that only one lock is needed? When calling the get/put/delete/update methods, the first step is to get the lock.

File: storage/minifile/mimifile.go (hunk 1 above)
Comment (sandyzhou) [reply storage/minifile/mimifile.go:65]: fixed

File: storage/minifile/mimifile.go (hunk 2 above)
Comment (sandyzhou) [reply storage/minifile/mimifile.go:120]: fixed

File: storage/minifile/mimifile.go (hunk 3 above)
Comment (sandyzhou) [reply storage/minifile/mimifile.go]: namedlock is removed.

File: storage/minifile/mimifile.go [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+	if mf.isClosed() {
+		return nil, fmt.Errorf("the miniFile storage is closed")
+	}
+
+	mf.lock.RLock()
+
+	val, err := mf.get(key)
+	if err != nil {
+		mf.lock.RUnlock()
+		_ = mf.Delete(key)
```
Comment (Lizen0512) [inline storage/minifile/mimifile.go]: mf.Delete(key) change to os.Remove(path), and then release the RUnlock by defer ？

File: storage/minifile/mimifile.go [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+	files, err := mf.prefix("")
+	if err != nil {
+		return nil, err
+	}
+
+	for _, file := range files {
+		val, err := mf.get(file)
+		if err != nil {
+			mf.lock.RUnlock()
+			_ = mf.Delete(file)
```
Comment (Lizen0512) [inline storage/minifile/mimifile.go]: Same as Get() method

File: storage/minifile/mimifile.go (hunk 4 above)
Comment (sandyzhou) [reply storage/minifile/mimifile.go]: fixed

File: storage/minifile/mimifile.go (hunk 5 above)
Comment (sandyzhou) [reply storage/minifile/mimifile.go]: fixed

**Review by Alexader — APPROVED** [review]

**Review by jzhe886 — APPROVED** [review]

**Pull request merged**